### PR TITLE
FIX: allow user to pick "eyegaze" or "pupil" (fix in triage_eyetrack_picks)

### LINKED ISSUE #12018: BUG: triage_eyetrack_pick does not work
in #11152 we intended to allow the user to pass `"eyetrack"`, `"eyegaze"`, or `"pupil"` to `inst.pick()`, but currently only `inst.pick("eyetrack")` works.

```
import mne
from mne.datasets.eyelink import data_path

fpath = data_path() / "sub-01_task-plr_eyetrack.asc"
raw = mne.io.read_raw_eyelink(fpath)

raw.copy().pick("eyetrack") # works
raw.copy().pick("pupil")  # FAILS
```

### PR DESCRIPTION
**Pull request by scott-huberty**
Fixes #12018

Now, `inst.pick("eyegaze")` or `inst.pick("pupil")` will work. 

EDIT: Note to self, verify that tests pass, then update changelog.

### CODE REVIEW

**Review by larsoner — APPROVED** [review]
Tests are passing so feel free to push the `devel.rst` update

File: mne/_fiff/pick.py [hunk 1]
```diff
@@ -1391,6 +1393,11 @@ def _picks_str_to_idx(
                 kwargs["fnirs"] = list(fnirs)[0]
             else:
                 kwargs["fnirs"] = list(fnirs)
+        if len(eyetrack) > 0 and not kwargs.get("eyetrack", False):
+            if len(eyetrack) == 1:
+                kwargs["eyetrack"] = list(eyetrack)[0]
+            else:
+                kwargs["eyetrack"] = list(eyetrack)
```
Comment (drammock) [inline mne/_fiff/pick.py:1400]: I have a bias against if/else constructs where the code in each branch is very very similar, so this suggestion is maybe DRY enough to become obfuscated?  So treat as an optional suggestion:
```suggestion
            idx = 0 if len(eyetrack) == 1 else slice(None)
            kwargs["eyetrack"] = list(eyetrack)[idx]
```
... but if you do it, it should also be propogated to the `fnirs` case to make clear that they're parallel / doing the same thing

**Review by drammock — APPROVED** [review]

File: mne/_fiff/pick.py (hunk 1 above)
Comment (scott-huberty) [reply mne/_fiff/pick.py:1400]: It's a little harder for _me_ to comprehend bc I wasn't aware of the `slice(None)` indexing trick,  but no strong feelings from my side, I can commit the suggestion for both fnirs and eyetrack

**Comment by larsoner** [comment]
Thanks @scott-huberty !

**Pull request merged**